packages/peregrine/lib/talons/RootComponents/Product/useProduct.js

```diff
@@ -0,0 +1,111 @@
+import { useEffect, useMemo } from 'react';
+import { useApolloClient, useLazyQuery } from '@apollo/react-hooks';
+
+/**
+ * A [React Hook]{@link https://reactjs.org/docs/hooks-intro.html} that
+ * controls the logic for the Product Root Component.
+ *
+ * @kind function
+ *
+ * @param {object} props
+ *
+ */
+export const useProduct = props => {
+    const {
+        fragment,
+        query,
+        urlKey // TODO: move `getUrlKey` into Peregrine?
+    } = props;
+
+    const onServer = false;
+
+    const fragmentVariables = useMemo(() => {
```

Comment:
IMO this is overuse of `useMemo` here and for `queryVariables` below. We should save it for when we have some expensive recalculation that could occur on every render. I'd just move these inline to where they are used. I don't think they gain us much by pre-defining, even outside the talon.

Reply:
Resolved in https://github.com/magento/pwa-studio/pull/1935/commits/cb6e9acf5fac29f6ecebfc34faabc63fc7e7854b.